WordPress/Classes/System/SplitViewRootPresenter.swift

```diff
@@ -361,17 +389,50 @@ extension SplitViewRootPresenter: UISplitViewControllerDelegate {
             sitePickerPopoverVC?.presentingViewController?.dismiss(animated: true)
         }
     }
-}
 
-extension SplitViewRootPresenter: SiteMenuViewControllerDelegate {
-    func siteMenuViewController(_ siteMenuViewController: SiteMenuViewController, showDetailsViewController viewController: UIViewController) {
-        if viewController is UINavigationController ||
-            viewController is UISplitViewController {
-            splitVC.setViewController(viewController, for: .secondary)
-        } else {
-            // Reset previous navigation or split stack
-            let navigationVC = UINavigationController(rootViewController: viewController)
-            splitVC.setViewController(navigationVC, for: .secondary)
+    func splitViewController(_ svc: UISplitViewController, willShow column: UISplitViewController.Column) {
+        if column == .primary, let selection = displayingContent?.selection, selection != sidebarViewModel.selection {
+            sidebarViewModel.selection = selection
```

Comment:
> the source of truth should always be sidebarViewModel.selection. 

That'd be my preference too. However, I think it's a bit hard to guarantee that. For example, any place can call the `configure(selection:)` function instead of `sidebarViewModel.selection = ...`. They should not do that, but they can.

If we'd like to make sure the data and view in-sync, it's probably best to make it impossible to programmatically updating view without updating the data.

Reply:
I'll update the PR to always use `sidebarViewModel.selection` to update UI. We can figure out the "make it impossible to programmatically updating view without updating the data" part later.